pkg/util/metrics/framework/registry.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/version"
+)
+
+// DefaultGlobalRegistry is a stub for the global registry which prometheus client
+// currently uses.
+var DefaultGlobalRegistry = NewKubeRegistry()
```

Comment:
we should start avoiding global variables. Each component should have such a registry.

Reply:
Yeah I agree. However it's mostly an orthogonal change to this particular effort. Definitely want to do it though.